components/prism-javascript.js

```diff
@@ -10,21 +10,31 @@ Prism.languages.insertBefore('javascript', 'keyword', {
 	'regex': {
 		pattern: /((?:^|[^$\w\xA0-\uFFFF."'\])\s])\s*)\/(\[[^\]\r\n]+]|\\.|[^/\\\[\r\n])+\/[gimyu]{0,5}(?=\s*($|[\r\n,.;})\]]))/,
 		inside: {
-			'regex-flag': {
-				pattern: /(\/)[gimyu]{1,5}$/,
-				lookbehind: true
-			},
-			'regex-charset1': {
-				pattern: /(\[(\\\]|[^\]])*\])|(\\[wds])/i,
-				alias: 'regex-charset'
-			},
-			'regex-escape': /\\(\\|\+|x[0-9a-f]{2}|u[0-9A-F]{4}|u\{\d+\}|c[a-z]|0[0-7]{1,2}|[^\d])/i,
-			'regex-reference': /\\\d/,
-			'regex-charset2': {
-				pattern: /\./,
-				alias: 'regex-charset'
+			'regex-flag': /[gimyu]{1,5}$/,
+			'regex-charset': {
+				pattern: /((?:^|[^\\])(?:\\\\)*)\[(?:\\.|[^\\\]])*\]/,
+				lookbehind: true,
+				inside: {
+					'regex-range': {
+						pattern: /((?:^|[^\\])(?:\\\\)*)\\?.-\\?./,
```

Comment:
This has problems with ranges like `\xA0-\uFFFF`.

To do this correctly one will have to look out for escape sequences, include them as well into `range` and then match them inside again as `escape`. 
Troublesome.

Maybe we want to leave that one out?
The pattern will be ridiculously long...